### PR TITLE
feat: add data elements to demo

### PR DESCRIPTION
**Pull request by czgu**
- also fixed some issues with string data element
- added tags and data element in demo
- hide data element search if it is not available

### CODE REVIEW

File: querybook/server/logic/data_element.py [hunk 1]
```diff
@@ -26,6 +26,14 @@ def get_data_element_by_name(name: str, session=None):
     return DataElement.get(name=name, session=session)
 
 
+@with_session
+def get_query_metastore_has_data_elements(metastore_id, session=None):
```
Comment (jczhong84) [inline querybook/server/logic/data_element.py:30]: the function name (expects metastores) seems not match what it actually returns (boolean)

File: querybook/server/logic/data_element.py [hunk 2]
```diff
@@ -112,25 +127,20 @@ def create_data_element_association(
...
 
+    data_element = None
+    if data_element_tuple:
+        if type(data_element_tuple) is str:
+            data_element = get_data_element_by_name(data_element_tuple, session=session)
+        elif data_element_tuple.name and data_element_tuple.type:
+            data_element = create_or_update_data_element(
+                metastore_id, data_element_tuple, session=session
+            )
+
```
Comment (jczhong84) [inline querybook/server/logic/data_element.py:143]: does it need to check the `data_element is None` again here?

File: querybook/webapp/components/Search/SearchOverview.tsx [hunk 3]
```diff
@@ -348,14 +348,11 @@ export const SearchOverview: React.FC<ISearchOverviewProps> = ({
...
-            loadEntities={DataElementResource.search}
-            onEntitiesChange={updateDataElements}
-            placeholder="search data element"
-        />
-    );
+    const queryMetastore =
+        searchType === SearchType.Table &&
+        queryMetastores.find((metastore) => metastore.id === metastoreId);
+    const queryMetastoreHasDataElements =
+        !!queryMetastore?.flags?.has_data_element;
```
Comment (jczhong84) [inline querybook/webapp/components/Search/SearchOverview.tsx:355]: nit: `queryMetastores.some((metastore) => metastore.id === metastoreId && !!metastore.flags?.has_data_element)`

File: querybook/server/logic/data_element.py (hunk 2 above)
Comment (czgu) [reply querybook/server/logic/data_element.py:143]: i added a check on line 148

**Review by jczhong84 — APPROVED** [review]

**Pull request merged**